xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-webjar/src/main/webjar/lightbox.js

```diff
@@ -93,8 +94,11 @@ define('xwiki-lightbox-description', [
     }
 
     if (attachmentData.date) {
+      // This should be updated after fixing XWIKI-19808: Proper javascript API to perform Date Formatting.
       var dateFormat = moment().toMomentFormatString($('.lightboxDescription .date').attr('data-dateFormat'));
-      $('.lightboxDescription .date').text(l10n.get('date', moment(attachmentData.date).format(dateFormat)));
+      var timezone = $('.lightboxDescription .date').attr('data-userTimezone');
+      $('.lightboxDescription .date')
+        .text(l10n.get('date', moment(attachmentData.date).tz(timezone).format(dateFormat)));
```

Comment:
This conversion to moment format strings seems to be from here: https://www.npmjs.com/package/moment-jdateformatparser/v/0.1.2 - though this library seems largely unmaintained and has bugs: https://github.com/MadMG/moment-jdateformatparser/issues/18 A more recent library is available [here](https://www.npmjs.com/package/@rangerrick/moment-javaformat) but I guess due to the API breakages this would cause it isn't possible for us to migrate, we could just use this in addition? Anyways, I think this clearly shows the need for implementing our own date formatting library in JavaScript that encapsulates all this logic ASAP to make sure that we can fix such issues/migrate to fixed libraries easily without having to change every usage.

Reply:
On the lightbox code I think the explicit conversion is not necessarily given that the `dateFormat` attribute should not be empty since a default value is provided here https://github.com/xwiki/xwiki-platform/blob/4ec93d34735e43489e7c10b358be58a082abfd33/xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-ui/src/main/resources/XWiki/Lightbox/WebHome.xml#L407
For the LD code I am not sure where is `this.config.dateFormat` initialized.

Probably switching to a new library should be taken into consideration while working on https://jira.xwiki.org/browse/XWIKI-19808

Reply:
No, this explicit conversion is not for the `null` case. This is for the fact that the placeholders that are supported in the format string in Java aren't the same as the ones in moment. The library `moment-jdateformatparser` provides a conversion between the two format string variants.

Reply:
Oh, I understood now your question. This conversion is done 2 lines above at https://github.com/xwiki/xwiki-platform/blob/4ec93d34735e43489e7c10b358be58a082abfd33/xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-webjar/src/main/webjar/lightbox.js#L98